mllib/src/main/scala/org/apache/spark/ml/classification/OneVsRest.scala

```diff
@@ -337,8 +353,13 @@ final class OneVsRest @Since("1.4.0") (
       paramMap.put(classifier.labelCol -> labelColName)
       paramMap.put(classifier.featuresCol -> getFeaturesCol)
       paramMap.put(classifier.predictionCol -> getPredictionCol)
-      classifier.fit(trainingDataset, paramMap)
-    }.toArray[ClassificationModel[_, _]]
+      Future {
+        classifier.fit(trainingDataset, paramMap)
+      }(executionContext)
+    }
+    val models = modelFutures
+      .map(ThreadUtils.awaitResult(_, Duration.Inf)).toArray[ClassificationModel[_, _]]
```

Comment:
@holdenk Because `ThreadUtils.awaitResult` wraps and re-throws any exceptions thrown by the underlying `Await` call, ensuring that this thread's stack trace appears in logs